src/ucp/rndv/rndv.c

```diff
@@ -821,6 +802,10 @@ ucp_rndv_recv_frag_put_mem_type(ucp_request_t *rreq, ucp_request_t *rndv_req,
                                       ucp_ep_use_indirect_id(freq->send.ep));
     }
 
+    ucs_assert(freq->send.ep != NULL);
+    ucs_hlist_add_tail(&ucp_ep_ext_gen(freq->send.ep)->proto_reqs,
```

Comment:
why do we need to save fragments on the list for put? should not transport report failure? otherwise the request should stay on pending, rigth?

Reply:
this is PUT operation from HOST staging buffer to GPU device buffer after receiving ATP from a peer on each fragment or after successful GET operation in pipelined RNDV
yes, we don't need this, since it is a local operation

removed